chapter06/src/main/java/com/seaofnodes/simple/node/IfNode.java

```diff
@@ -27,14 +27,14 @@ StringBuilder _print1(StringBuilder sb) {
     @Override
     public Type compute() {
         // If the If node is not reachable then neither is any following Proj
-        if (ctrl()._type != Type.CONTROL) return TypeTuple.IF_BOTH_UNREACHABLE;
+        if (ctrl()._type != Type.CONTROL) return TypeTuple.IF_NEITHER;
         // If test os not a constant true/false then assume both branches are reachable
-        if (!pred()._type.isConstant())   return TypeTuple.IF_BOTH_REACHABLE;
+        if (!pred()._type.isConstant())   return TypeTuple.IF_BOTH;
         // If constant is 0 then false branch is reachable
         // Else true branch is reachable
         if (pred()._type instanceof TypeInteger ti) {
```

Comment:
I think this is missing a `ti.isConstant()` check

Reply:
Nothing else returns true for isConstant - but yes safer to not assume I think

Reply:
The type could be int top or int bottom which are not constants but would here result in such.